src/ContentfulWrapper.js

```diff
@@ -15,4 +16,43 @@ const getPublishedSuppliers = async () => {
   });
 };
 
-export { getPublishedSuppliers };
+const updateSupplier = async (pair) => {
+  const client = contentful.createClient({
+    accessToken: import.meta.env.VITE_REACT_APP_CMA_TOKEN,
```

Comment:
I don't really understand how Contentful apps work, but does this hard code the CMA token at build time? (is this a security concern?)

Reply:
> does this hard code the CMA token at build time

I've been thinking about this too.  Ultimately it's all built into client-side js, so yes. But it's only run in Contentful, so you will have had to have authenticated to access that JS anyway.  I guess a more likely concern is something like someone getting access to the built images in GitHub Actions.  They aren't persisted and I don't know how that could happen but that doesn't mean it couldn't.

I might ask Contentful what they recommend - this is the approach in their docs so I have just assumed it's ok.

Their docs do mention app keys and signed requests, but these are for backend apps, ie an app running on our servers that interacts with Contentful.  Either way, I'll make a service account for this app and limit what the key can do.

Reply:
Found the `cmaAdapater` - https://github.com/citizensadvice/contentful-csr-upload/pull/55

So I can initialise the `client` without providing an app token 🙌🏻. Thanks for this!

Reply:
that makes much more sense, I would have assumed a Contentful app is intended to operate as the user that is logged in rather than a service account so I would have hoped it had a feature like that!